src/app/dim-ui/Sheet.tsx

```diff
@@ -226,9 +226,14 @@ export default function Sheet({
       if (sheetContents.current.clientHeight > 0) {
         setFrozenHeight(sheetContents.current.clientHeight);
       } else {
-        setTimeout(() => {
-          sheetContents.current && setFrozenHeight(sheetContents.current.clientHeight);
-        }, 500);
+        const setHeight = () => {
+          if (!sheetContents.current || sheetContents.current.clientHeight === 0) {
+            return false;
+          }
+          setFrozenHeight(sheetContents.current.clientHeight);
+          return true;
+        };
+        tryWithBackoff(setHeight);
```

Comment:
I think this is just a slightly better version of the intended functionality. I don't think it is explicitly needed for this fix, but it was something I noticed upon having a read of the code.

Reply:
Just realised we need cancelation functionality to avoid race conditions of layout effects rerunning, so maybe I should just revert?